src/App.tsx

```diff
@@ -1,14 +1,49 @@
+/* eslint-disable import/no-extraneous-dependencies */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import debounce from 'lodash.debounce';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState(todos[0]);
+
+  const applyQuery = useCallback(debounce(setApliedQuery, 300), []);
+
+  useEffect(() => {
+    // setLoading(true);
+
+    if (status === 'active') {
+      getTodos()
+        .then(prevTodos => prevTodos.filter(todo => todo.completed === false))
+        .then(setTodos);
+    }
+
+    if (status === 'completed') {
+      getTodos()
+        .then(prevTodos => prevTodos.filter(todo => todo.completed === true))
+        .then(setTodos);
+    }
+
+    if (status === 'all') {
+      getTodos().then(setTodos);
+    }
+  }, [status]);
```

Comment:
Consider changing this logic, you need to fetch todos only once in your first render (empty dependencies array). And create some function to filter todos based on your status